features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -57,188 +272,1561 @@ const useStyles = makeStyles((theme) => ({
  */
 export default function Environments() {
     const classes = useStyles();
+    const intl = useIntl();
     const { api, updateAPI } = useContext(APIContext);
     const { settings } = useAppContext();
-    const [gatewayEnvironments, setGatewayEnvironments] = useState([...api.gatewayEnvironments]);
+    let revisionCount;
+    if (Configurations.app.revisionCount) {
+        revisionCount = Configurations.app.revisionCount;
+    } else {
+        revisionCount = 5;
+    }
     const [selectedMgLabel, setSelectedMgLabel] = useState([...api.labels]);
-    const [isUpdating, setUpdating] = useState(false);
     const [selectedDeployments, setSelectedDeployments] = useState([...api.deploymentEnvironments]);
-
     const restApi = new API();
     const [allDeployments, setAllDeployments] = useState([]);
+    const [allRevisions, setRevisions] = useState(null);
+    const [allEnvRevision, setEnvRevision] = useState(null);
+    const [selectedRevision, setRevision] = useState(null);
+    const [extraRevisionToDelete, setExtraRevisionToDelete] = useState(null);
+    const [description, setDescription] = useState('');
+    const [mgLabels, setMgLabels] = useState([]);
+    const [SelectedEnvironment, setSelectedEnvironment] = useState([]);
+    const [open, setOpen] = useState(false);
+    const [confirmDeleteOpen, setConfirmDeleteOpen] = useState(false);
+    const [revisionToDelete, setRevisionToDelete] = useState([]);
+    const [confirmRestoreOpen, setConfirmRestoreOpen] = useState(false);
+    const [revisionToRestore, setRevisionToRestore] = useState([]);
+    const [openDeployPopup, setOpenDeployPopup] = useState(false);
+    const [lastRevisionCount, setLastRevisionCount] = useState(0);
+
+    const extractLastRevisionNumber = (list, lastRev) => {
+        if (lastRev !== null) {
+            setLastRevisionCount(lastRev);
+        } else if (list[list.length - 1]) {
+            const lastRevisionName = list[list.length - 1].displayName;
+            const splitList = lastRevisionName.split(' ');
+            setLastRevisionCount(splitList[1]);
+        } else {
+            setLastRevisionCount(0);
+        }
+    };
+
     useEffect(() => {
         restApi.getDeployments()
             .then((result) => {
                 setAllDeployments(result.body.list);
             });
+        restApi.microgatewayLabelsGet()
+            .then((result) => {
+                setMgLabels(result.body.list);
+            });
+        restApi.getRevisions(api.id).then((result) => {
```

Comment:
We are making 2 API calls to get revisions, one in `APIDetailsTopMenue` and one in `Environments` component

![image](https://user-images.githubusercontent.com/3313885/105882311-82d83600-602b-11eb-84c9-8ebdc2f25125.png)

This is both redundant and error-prone and that's why we can't update the dropdown
![image](https://user-images.githubusercontent.com/3313885/105882435-a602e580-602b-11eb-93f4-e8194a9d9dcf.png)


when a new revision is created,
Shall we reduce this to 1 call ?